### PR TITLE
Edge Example App: Sense the Internet Part 1

### PR DESCRIPTION
**Pull request by ahmadateya**

### CODE REVIEW

File: _posts/2022-09-04-edge-example-app-sense-the-internet-part-1.markdown [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+- workloads
+
+summary-1:
+---
+Edge Example App is an app for Flotta Edge devices, with a workload that will be deployed on the device that has 2 main features:
+- Sensing the Internet (which helps to construct devices network topology). <= this article
+- Read CPU temperature (which indicate how much load the device is handling).
+
+As Project Flotta goal is to manage workloads deployed on small-footprint devices and dealing with network connectivity issues, this app provides a way of collecting network information that will be exported to the edge cluster.
+this App Along with the Web UI Interface, provides a good view of the network topology and the devices that are connected to the devices network.
```
Comment (gciavarrini) [inline _posts/2022-09-04-edge-example-app-sense-the-internet-part-1.markdown:19]: s/this/This/

not sure about the meaning of this sentence:
> the devices that are connected to the devices network"

 what is the devices network? The cluster? Internet network? Can you please reword it a bit to make it easier to understand?

File: _posts/2022-09-04-edge-example-app-sense-the-internet-part-1.markdown [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+
+## How this app works
+taking the simple way of getting network information, this app collects its information about the network by sending a packet on the IP network to the Edge Cluster, calculating the time taken for each hop the packet makes during its route to the destination, this much like traceroute in the network but done on the device.
+
+![](/assets/images/traceroute.png)
+
+as a workload app, this app benefits from Flotta Edge devices architecture, collecting its information and saving it to a file on the device and let the device worker send it to the edge cluster, this way
```
Comment (gciavarrini) [inline _posts/2022-09-04-edge-example-app-sense-the-internet-part-1.markdown:26]: s/as/As/

File: _posts/2022-09-05-edge-example-app-cpu-temp.markdown [hunk 3]
```diff
@@ -0,0 +1,50 @@
+---
+layout: post
+title:  "Edge Example App: CPU Temperature"
+date:   2022-09-05 03:27:05 +0100
```
Comment (gciavarrini) [inline _posts/2022-09-05-edge-example-app-cpu-temp.markdown:4]: This is another blog post, I'd like to suggest to open a separate PR

File: _posts/2022-09-04-edge-example-app-sense-the-internet-part-1.markdown [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+categories: flotta
+author: Ahmad Ateya <ahmad.m.ateya@gmail.com>
+tags:
+- flotta
+- devices
+- workloads
+
+summary-1:
+---
+Edge Example App is an app for Flotta Edge devices, with a workload that will be deployed on the device that has 2 main features:
```
Comment (gciavarrini) [inline _posts/2022-09-04-edge-example-app-sense-the-internet-part-1.markdown:14]: nit: maybe s/2/two/

File: _posts/2022-09-05-edge-example-app-cpu-temp.markdown [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+tags:
+- flotta
+- devices
+- workloads
+
+summary-1:
+---
+Edge Example App is an app for Flotta Edge devices, with a workload that will be deployed on the device that has 2 main features:
+
+Sensing the Internet (which helps to construct devices network topology). (see [part 1]() and [part 2]())
```
Comment (gciavarrini) [inline _posts/2022-09-05-edge-example-app-cpu-temp.markdown:16]: Do not forget: before merging the links should be filled

File: _posts/2022-09-05-edge-example-app-cpu-temp.markdown [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+Read CPU temperature (which indicate how much load the device is handling).<= this article
+
+As Project Flotta goal is to manage workloads deployed on small-footprint devices, reading the CPU temperature is a good way to know how much load the device is handling. this App Along with the Web UI Interface, helps to collect and visualize the CPU temperature of the device over the day by presenting it into Line Graph, making it easy to identify the time of the day when the device is handling more load.
```
Comment (gciavarrini) [inline _posts/2022-09-05-edge-example-app-cpu-temp.markdown:19]: `s/. this/. This/`

File: _posts/2022-09-05-edge-example-app-cpu-temp.markdown [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+
+
+## How this app works
+
+This app reads the CPU temperature from the device and saves it to a file on the device, the device worker will send it to the edge cluster, and the edge cluster will save it to the S3 bucket.
+
+in the backend side, the edge cluster will read the file from the S3 bucket, calculate the average temperature for every hour and expose this data to the web UI with endpoints, which will present it to the user into line graph.
```
Comment (gciavarrini) [inline _posts/2022-09-05-edge-example-app-cpu-temp.markdown:26]: `s/in/In/

not sure: "into line graph" or "as line graph"? can you check?

File: _posts/2022-09-05-edge-example-app-cpu-temp.markdown [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+2022/09/04 12:14:25 {"Temp":43}
+2022/09/04 13:15:25 {"Temp":44}
+```
+
+## Example of presented data
+<img src="/assets/images/cpu-temp-graph.png" alt="CPU graph line" width="800"/>
+
+## How to use this app
+
+this feature is part of the Edge Example App, so you can use the same manifests explained here part 1 to deploy the app to your device.
```
Comment (gciavarrini) [inline _posts/2022-09-05-edge-example-app-cpu-temp.markdown:42]: `s/this/This/`

Comment (gciavarrini) [reply _posts/2022-09-05-edge-example-app-cpu-temp.markdown:42]: > here part 1

missing link?

File: _posts/2022-09-04-edge-example-app-sense-the-internet-part-1.markdown (hunk 1 above)
Comment (pkliczewski) [inline _posts/2022-09-04-edge-example-app-sense-the-internet-part-1.markdown:19]: Do you think it makes sense to add links to repos?

File: _posts/2022-09-05-edge-example-app-cpu-temp.markdown [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+
+Sensing the Internet (which helps to construct devices network topology). (see [part 1]() and [part 2]())
+Read CPU temperature (which indicate how much load the device is handling).<= this article
+
+As Project Flotta goal is to manage workloads deployed on small-footprint devices, reading the CPU temperature is a good way to know how much load the device is handling. this App Along with the Web UI Interface, helps to collect and visualize the CPU temperature of the device over the day by presenting it into Line Graph, making it easy to identify the time of the day when the device is handling more load.
+
+
+## How this app works
+
+This app reads the CPU temperature from the device and saves it to a file on the device, the device worker will send it to the edge cluster, and the edge cluster will save it to the S3 bucket.
```
Comment (pkliczewski) [inline _posts/2022-09-05-edge-example-app-cpu-temp.markdown:24]: I would mentioned that this information could be sent to Prometheus or Thanos but to keep it aligned with `sensing the internet` we use the same data synchronization mechanism.

File: _posts/2022-09-05-edge-example-app-cpu-temp.markdown (hunk 3 above)
Comment (ahmadateya) [reply _posts/2022-09-05-edge-example-app-cpu-temp.markdown:4]: I made a PR [here](https://github.com/project-flotta/project-flotta.github.io/pull/113)

File: _posts/2022-09-05-edge-example-app-cpu-temp.markdown (hunk 5 above)
Comment (ahmadateya) [reply _posts/2022-09-05-edge-example-app-cpu-temp.markdown:16]: fixed them [here](https://github.com/project-flotta/project-flotta.github.io/pull/113)

File: _posts/2022-09-05-edge-example-app-cpu-temp.markdown (hunk 3 above)
Comment (pkliczewski) [reply _posts/2022-09-05-edge-example-app-cpu-temp.markdown:4]: Thanks, please remove it from here

File: _posts/2022-09-05-edge-example-app-cpu-temp.markdown (hunk 9 above)
Comment (ahmadateya) [reply _posts/2022-09-05-edge-example-app-cpu-temp.markdown:24]: If I got some time after GSoC, I would like to contribute to flotta projects and so I will add this to my list :)

File: _posts/2022-09-04-edge-example-app-sense-the-internet-part-1.markdown (hunk 1 above)
Comment (ahmadateya) [reply _posts/2022-09-04-edge-example-app-sense-the-internet-part-1.markdown:19]: I already added it at the end of the article, Do you think adding it inside the blogpost is more suitable?

File: _posts/2022-09-05-edge-example-app-cpu-temp.markdown (hunk 7 above)
Comment (ahmadateya) [reply _posts/2022-09-05-edge-example-app-cpu-temp.markdown:26]: as a line graph is better (Y)

File: _posts/2022-09-05-edge-example-app-cpu-temp.markdown (hunk 6 above)
Comment (ahmadateya) [reply _posts/2022-09-05-edge-example-app-cpu-temp.markdown:19]: done

File: _posts/2022-09-05-edge-example-app-cpu-temp.markdown (hunk 8 above)
Comment (ahmadateya) [reply _posts/2022-09-05-edge-example-app-cpu-temp.markdown:42]: done

File: _posts/2022-09-04-edge-example-app-sense-the-internet-part-1.markdown (hunk 1 above)
Comment (ahmadateya) [reply _posts/2022-09-04-edge-example-app-sense-the-internet-part-1.markdown:19]: Oh sorry I mean the internet network (the route from the edge device to the cluster), I will clarify it

File: _posts/2022-09-05-edge-example-app-cpu-temp.markdown (hunk 3 above)
Comment (ahmadateya) [reply _posts/2022-09-05-edge-example-app-cpu-temp.markdown]: done

File: _posts/2022-09-05-edge-example-app-cpu-temp.markdown (hunk 9 above)
Comment (gciavarrini) [reply _posts/2022-09-05-edge-example-app-cpu-temp.markdown]: I think you can still add the info suggested by Piotr. 
Project flotta already supports Prometheus or Thanos ([see](https://project-flotta.io/flotta/2022/04/11/writing-metrics-to-control-plane.html). You can also add some future work ideas

Comment (ahmadateya) [reply _posts/2022-09-05-edge-example-app-cpu-temp.markdown]: sure, will add this as a subsection in the post

File: _posts/2022-09-04-edge-example-app-sense-the-internet-part-1.markdown (hunk 4 above)
Comment (ahmadateya) [reply _posts/2022-09-04-edge-example-app-sense-the-internet-part-1.markdown]: done

File: _posts/2022-09-04-edge-example-app-sense-the-internet-part-1.markdown (hunk 2 above)
Comment (ahmadateya) [reply _posts/2022-09-04-edge-example-app-sense-the-internet-part-1.markdown]: done

**Review by pkliczewski — COMMENTED** [review]
Looks good. Please let me know when ready.

**Comment by pkliczewski** [comment]
@ahmadateya please fix `CheckLinks` issues

**Comment by ahmadateya** [comment]
the `CheckLinks` issues is related to the Part 2 article because I reference it here, should I comment it for now and open another PR to it after merging Part 2?

**Comment by ahmadateya** [comment]
the `CheckLinks` issue is related to the Part 1 and Part 2 articles because I reference them here, should I comment those lines for now and open another PR to it after merging Part 1 and Part 2 articles?

**Comment by pkliczewski** [comment]
If missing part 2 is the only reason it is OK. We will merge both PRs at the same time.

**Review by pkliczewski — APPROVED** [review]
@gciavarrini are you ok with merging this PR?

**Review by gciavarrini — APPROVED** [review]

**Pull request merged**